lib/src/features/leaderboard/ui/leaderboard_widget.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:lichess_mobile/src/features/leaderboard/data/leaderboard_repository.dart';
+import 'package:lichess_mobile/src/widgets/feedback.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+import 'package:lichess_mobile/src/common/lichess_icons.dart';
+
+import '../model/leaderboard.dart';
+import './leaderboard_screen.dart';
+
+final leaderListProvider = FutureProvider.autoDispose((ref) async {
+  final leaderRepo = ref.watch(leaderboardRepositoryProvider);
+  final either = await leaderRepo.getLeaderboard().run();
+  return either.match((error) => throw error, (data) {
+    ref.keepAlive();
+    return data;
+  });
+});
+
+class LeaderboardWidget extends ConsumerStatefulWidget {
+  const LeaderboardWidget({super.key});
+
+  @override
+  ConsumerState<LeaderboardWidget> createState() => _LeaderboardWidgetState();
+}
+
+class _LeaderboardWidgetState extends ConsumerState<LeaderboardWidget> {
```

Comment:
there is no field in your state, so you don't need the `StatefulWidget` here and can use a `ConsumerWidget` instead.